mediator/src/main/scala/io/iohk/atala/mediator/protocols/PickupExecuter.scala

```diff
@@ -90,7 +90,7 @@ object PickupExecuter
             from = m.to.asFROM,
             to = m.from.asTO,
             recipient_did = m.recipient_did,
-            attachments = messagesToReturn.map(m => (m.hashCode.toString, m.msg)).toMap,
+            attachments = messagesToReturn.map(m => (m._id, m.msg)).toMap,
```

Comment:
@mineme0110 I think this was another bug
`m` is the MessageItem, not the Message
We need to double-check the attachment id and the delete of messages by that ID (mark as read).